azerothcore/azerothcore-wotlk
# fix(Scripts/ScarletMonastery): improvement ScarletMonastery

## Changes Proposed:

1. Go to the pull request #15218 from @feltwitch to see the detailed changes I gave with the sniffing data given by @heyitsbench based on that pull request Reworked triggering the Ember Bringer event.
2. The door should open after the event is triggered, and Mograini should be able to receive a welcome dialogue. 
3. NPCs inside the church. All Friendly, Removed Whiteman

4 . Added the emote（salute） of Whitemane's creature after resurrecting Mograine（growl）. 
5. After resurrecting Mograine, Whitemayne dies, the player quits the fight or dies, and Mograine does not respawn Pick up Mograine's items without a shot
Out of combat should be respawning rather than waiting for the mob to move back to the spawn point
6. Whitemane's HOLY_SMITE spells should be cast at ranged (not moving) whenever possible, and will only attempt to move to melee range if the spell is unusable (behind a building, etc.).
7. Whitemane has less than 50% HP and is released DEEP_SLEEP after the spell should be moved to Mograine's side before using Resurrection (if the player does not have a DEEP_SLEEP debuff after moving to Mograine, the resurrection will be released immediately, and if there is a debuff, it will wait for 5 seconds before it is released)
8. When Mograine has dot damage and triggers the first death, the dot damage number can still appear on Mograine, and does not remove all Aura


This PR proposes changes to:

## Issues Addressed:

- Closes 
https://github.com/azerothcore/azerothcore-wotlk/issues/14478

## SOURCE:

The changes have been validated through:

https://github.com/azerothcore/azerothcore-wotlk/pull/15218

## Tests Performed:

This PR has been:

## How to Test the Changes:



#### Tests Ashbringer

1. `.additem 22691`.
2. `.level 60` or `.character level 60`.
3. Armament weapon ID 22691 `.additem 22691`.
4. `.instance unbind all`.
5. `.tele smcath`.
6. The doors of the church should be opened.
7. Unloading weapon ID 22691 after Ashbringer event All creatures in the dungeon should be friendly (including in churches)
#### Test combat

1. `.level 80` or `.character level 80`.
2. It is recommended to use a Warlock or Priest to test the battle when changing weapons.
3. Use dot spells on Mograine as much as possible, such as Doomsday Curse, Corrosion, etc. After Mograine dies for the first time and falls to the ground, you can still see the damage on Mograine on the ground.
4. Whitemane exits from behind the gate and moves to Mograine's side, and does not enter the battle (at player level 80, enters the battle at low level), at which point Whitemane will always stay outside the gate if .gm on is used, and will not return to the spawn point.
5. Whitemane should always cast HOLY_SMITE spell in place, and after taking a detour behind a pillar, Whitemane will attempt to move to melee range to attack the player
6. After Whitemane first enters the fight Pull him away from Mograine's corpse (the farther away the better) and reduce his HP to less than 50% Trigger the Resurrection Mograine event Whitemane hypnotizes the player and starts moving next to Mograine's corpse (now just stays in place and uses the resurrection spell until Mograine is revived) Until Mograine resurrects, his HP will only drop to 1 at most, and he will not be able to kill him
7. Whitemane will make an emote after resurrecting Mograine

## Known Issues and TODO List:


- [x] Adds missing behaviors in the resurrection event when Mograine and Whitemane fight
- [x] Modified Whitemane act of releasing spells
- [x] Move C++ scripts to SmartScript

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

> http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
